ocaml/xapi/xapi_pool.ml

```diff
@@ -800,7 +800,13 @@ let emergency_transition_to_master ~__context =
 let emergency_reset_master ~__context ~master_address =
   if Localdb.get Constants.ha_armed = "true" 
   then raise (Api_errors.Server_error(Api_errors.ha_is_enabled, []));
+  if Pool_role.is_master ()
+  then raise Api_errors.(Server_error(host_is_already_master, []));
```

Comment:
I think that `host_is_master` (from the previous version of the patch) is better. This call is not to make the host a master, but to update a slave's view of where the master is.